test/core.jl

```diff
@@ -7973,3 +7973,6 @@ let spec = only(methods(g47476)).specializations
     @test any(mi -> mi !== nothing && Base.isvatuple(mi.specTypes), spec)
     @test all(mi -> mi === nothing || !Base.has_free_typevars(mi.specTypes), spec)
 end
+
+f48950(::Union{c,Nothing}...) where {c,d} = 1
+@test f48950(1, 1) == 1
```

Comment:
```suggestion
f48950(::Union{Int,d}, ::Union{c,Nothing}...) where {c,d} = 1
@test f48950(1, 1, 1) == 1
```
this ensures it is a legal definition (no warning), but still hits the same code path